app/scripts/contentscript.js

```diff
@@ -297,9 +297,14 @@ const setupPageStreams = () => {
   pageChannel = pageMux.createStream(PROVIDER);
 };
 
+// The field below is used to ensure that replay is done only once for each restart.
+let METAMASK_EXTENSION_CONNECT_SENT = false;
+
 const setupExtensionStreams = () => {
+  METAMASK_EXTENSION_CONNECT_SENT = true;
```

Comment:
nit: maybe `METAMASK_EXTENSION_CONNECT_SENT` -> `METAMASK_EXTENSION_CAN_SEND_RETRY`? thoughts? 
 
cc: @naugtur

Reply:
Agreed that calling this "connect sent" isn't idea, because it doesn't really represent whether the extension has connected. After we send the retry command, it remains true that the extension has connected, but this is `false`.

Maybe instead of "can send retry", it should be "should send retry"? Or "send retry when ready"

OR maybe it would be simpler to flip this around. Use this boolean to track when the retry has already been sent, call it "METAMASK_EXTENSION_RETRY_SENT", and set  it to `false` here. Then we can set to `true` in `extensionStreamMessageListener`

Reply:
How about we call it what it is - a semaphore. 

`METAMASK_EXTENSION_CONNECT_SEMAPHORE`